### DISCUSSION TITLE
Polaris should be a simple CSS Admin UI Framework. like Bootstrap, Tailwind CSS, or Foundation.

### DISCUSSION THREAD

**Discussion by javmah** [Shopify/polaris-react#9826] [Ideas]
Polaris should be a simple CSS Admin UI Framework. like Bootstrap, Tailwind CSS, or Foundation. the web is changing, new technologies come and go. today's best is tomorrow's worst. Basic things should be the same. Forcing people to use React is not a Great Idea. Suppose you are going to use a Polaris table component and your table is Complex with different data uses you can't do that with a Polaris table component! LOL. Even if you add Polaris CSS with CDN  you have to add wired classes along with 400 % more HTML. Ther is no room for Solid.js, svelte or Vue devolapers. It's a serious issue Polaris Degine system should be CSS and simple js. like other UI frameworks.  

"Any damn fool can make it complex. It takes a genius to make it simple. — Ray Dalio"

**Comment by sam-b-rose** [comment]
@javmah We are currently exploring this path! Thank you for adding this as a discussion topic.

**Comment by juzser** [comment]
Totally agree. 
Our company is using Vue as main framework for all projects. And we see Vue is better than React (Performance, learning curve, cost to dev, time to dev,...) and the most important is Vue code/structure is much easier to look at than React. 
While Shopify releases Polaris and all the libs based on React, the our dev team is sad, and I see around in the Vue community, they're not happy too...

Our team had to pick some guys to dev a Vue version of Polaris [here](https://github.com/ownego/polaris-vue).
Luckily, it's really good and we gain Built for Shopify badges for almost all of our apps. 

Hopefully Shopify can look more deeply and listen to the dev community. 

![vue-open-source](https://github.com/Shopify/polaris/assets/5735071/83dcf450-dd77-4996-9e04-ebaadcf42828)

**Reply by bespokely-digital** [reply]
Yes, we too are using Vue for current and future apps and using the same npm package for now. Removing the react complexity would be fab.

**Comment by dizzyjaguar** [comment]
Any update on this? I keep running into issues using the components from Polaris that don't have all of the props that plain html elements don't have. For example I need access to `name` and `value` on **Button** but that doesn't appear to possible.

**Reply by saneruby** [reply]
> Any update on this? I keep running into issues using the components from Polaris that don't have all of the props that plain html elements don't have. For example I need access to `name` and `value` on **Button** but that doesn't appear to possible.

Same situation, it is real problem

**Comment by alexandrosk** [comment]
I'm recreating Polaris components using tailwind and shadcn as a base will try to OS as soon as I'm done 

If anyone finishes this earlier lmk

**Reply by khrnchn** [reply]
bro are you still cooking? i need it so bad

**Comment by daviareias** [comment]
I think another problem is that the CSS variables names are really big, if you're only using only CSS this is ok, however I noticed most react components add those classes dynamically making some components even bigger than they already are.

**Comment by DebanjanT** [comment]
# 🎉 I Created a New Library: **Tailwind Polaris**

I’m excited to share that I’ve created a new library called **Tailwind Polaris**! 🚀

**Tailwind Polaris** is a Tailwind CSS implementation of the Shopify Polaris design system. It provides a comprehensive set of components and utilities to help developers build modern, accessible, and responsive user interfaces with ease.

---

## 📦 NPM Package

You can find the library on NPM here:  
👉 [@dtewary/tw-polaris](https://www.npmjs.com/package/@dtewary/tw-polaris)

---

## ✨ Features

- Fully customizable components built with **Tailwind CSS**.
- Implements the **Shopify Polaris Design System**.
- Includes modern, accessible, and responsive UI components.
- Supports **Radix UI** for advanced components like `Accordion`, `Dialog`, `Tooltip`, and more.
- Adheres to **Semantic Versioning** for reliable updates.

---

## 🔗 Links

- **NPM Package**: [@dtewary/tw-polaris](https://www.npmjs.com/package/@dtewary/tw-polaris)  
- **Storybook**: [Tailwind Polaris Storybook](https://dub.sh/twpolaris)  
- **Website**: [Tailwind Polaris Website](https://polaris.dtewary.com) (Development on hold)

**Comment by daviareias** [comment]
I'm sorry for disappointing anyone trying to use polaris components with pure CSS, but except for a few components that obviously don't need javascript such as `Card`, `Grid` , `Inline stack` and `Box`, a lot of elements such as `Autocomplete` or `Radio Button` are not possible to be built using CSS only if you wan't to adhere to accessibility requirements.

**Comment by DebanjanT** [comment]
Actually I am not working on it now you can do that I will check it np
